configuration/configuration.go

```diff
@@ -356,6 +359,16 @@ func (gc *GlobalConfiguration) initTracing() {
 
 func (gc *GlobalConfiguration) initACMEProvider() {
 	if gc.ACME != nil {
+		if len(gc.ACME.CAServer) == 0 {
+			gc.ACME.CAServer = DefaultAcmeCAServer
+		}
+
+		if strings.Contains(gc.ACME.CAServer, "v01") {
+			caServer := strings.Replace(gc.ACME.CAServer, "v01", "v02", 1)
+			log.Warn("the CA server %q refer to a v01 endpoint of the ACME API, please change to %q.", gc.ACME.CAServer, caServer)
```

Comment:
The log entry doesn't mention that you swapped v01 for v02. it is possible that for some reason, someone may want to use a CA server that contains `v01` in the URL. I would test for a more exact match, or just warn in the logs.